Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -3416,6 +3416,12 @@ void RenderLayer::paintLayerContents(GraphicsContext& context, const LayerPainti
 
 void RenderLayer::paintLayerByApplyingTransform(GraphicsContext& context, const LayerPaintingInfo& paintingInfo, OptionSet<PaintLayerFlag> paintFlags, const LayoutSize& translationOffset)
 {
+    auto localPaintFlags = paintFlags;
+    if (m_renderer.isDocumentElementRenderer()) {
+        paintLayerContentsAndReflection(context, paintingInfo, paintFlags | PaintLayerFlag::PaintingRootBackgroundOnly | PaintLayerFlag::TemporaryClipRects);
```

Comment:
Doesn't this recur through all descendant layers, though?

Reply:
Indeed, it probably will :(

Do you know what the best way to solve this is?

Possibly we only want to do this if we're in the right paint phase, and strip out other PaintFlagFlags to ensure that we don't iterate descendants? (Is it just `PaintingCompositingBackgroundPhase`, or are we not always compositing for the root)?